rts/c/cuda.h

```diff
@@ -249,6 +257,45 @@ static int cuda_device_setup(struct cuda_context *ctx) {
     fprintf(stderr, "Using device #%d\n", chosen);
   }
 
+  int used_devices = 0;
+  int* devices = (int*)malloc(sizeof(int)*count);
+  for(int devID = 0; devID < count; devID++){
+    CUDA_SUCCEED_FATAL(cuDeviceGet(&dev, devID)); 
+
+    int cc_major = device_query(dev, COMPUTE_CAPABILITY_MAJOR);
+    int cc_minor = device_query(dev, COMPUTE_CAPABILITY_MINOR);
+    // Here it's a design chocie that all device should be of the same CC
+    if (cc_major == cc_major_best && cc_minor == cc_minor_best){
+      devices[used_devices] = devID;
+      used_devices++;
+    }
+  }
+  if(ctx->cfg.debugging){
+    fprintf(stderr, "Using %d devices\n", used_devices);
+  }
+  ctx->kernel_iterator = 0;
+  ctx->device_count = used_devices;
+  ctx->devices = (CUdevice*)malloc(sizeof(CUdevice)*used_devices);
+  ctx->contexts = (CUcontext*)malloc(sizeof(CUcontext)*used_devices);
+  ctx->modules = (CUmodule*)malloc(sizeof(CUmodule)*used_devices);
+  ctx->kernel_done = (CUevent*)malloc(sizeof(CUevent)*used_devices*2);
+  ctx->finished = (CUevent*)malloc(sizeof(CUevent)*used_devices);
+
+  for(int devIdx = 0; devIdx < used_devices; devIdx++){
+    CUDA_SUCCEED_FATAL(cuDeviceGet(&ctx->devices[devIdx], devices[devIdx]));
+    CUDA_SUCCEED_FATAL(cuCtxCreate(&ctx->contexts[devIdx], 
```

Comment:
The documentation for `cuCtxCreate()` suggests that it is better to use `cuDevicePrimaryCtxRetain()`.